solnlib/conf_manager.py

```diff
@@ -557,3 +550,68 @@ def get_log_level(
             f"taking {default_log_level} as log level."
         )
         return default_log_level
+
+
+def get_proxy_dict(
+    logger: logging.Logger,
+    session_key: str,
+    app_name: str,
+    conf_name: str,
+    proxy_stanza: str = "proxy",
+    **kwargs,
+) -> Union[Dict[str, str], NoReturn]:
+    """This function returns the log level for the addon from configuration
+    file.
+
+    Arguments:
+        logger: Logger.
+        session_key: Splunk access token.
+        app_name: Add-on name.
+        conf_name: Configuration file name where logging stanza is.
+        proxy_stanza: Proxy stanza that would contain the Proxy details
+    Returns:
+        A dictionary is returned with stanza details present in the file.
+        The keys related to `eai` are removed before returning.
+
+    Examples:
+        >>> from solnlib import conf_manager
+        >>> proxy_details = conf_manager.get_proxy_dict(
+        >>>     logger,
+        >>>     "session_key",
+        >>>     "ADDON_NAME",
+        >>>     "splunk_ta_addon_settings",
+        >>> )
+    """
+    proxy_dict = {}
+    try:
+        cfm = ConfManager(
+            session_key,
+            app_name,
+            realm=f"__REST_CREDENTIAL__#{app_name}#configs/conf-{conf_name}",
+        )
+        conf = cfm.get_conf(conf_name)
+    except ConfManagerException:
+        logger.error(f"Failed to fetch configuration file '{conf_name}'.")
+    else:
+        try:
+            proxy_dict = conf.get(proxy_stanza)
+        except ConfStanzaNotExistException:
+            logger.error(
+                f"Failed to fetch '{proxy_stanza}' from the configuration file '{conf_name}'. "
+            )
+        else:
+            # remove the other fields that are added by ConfFile class
+            proxy_dict.pop("disabled", None)
+            proxy_dict.pop("eai:access", None)
+            proxy_dict.pop("eai:appName", None)
+            proxy_dict.pop("eai:userName", None)
+
+            if "proxy_port" in kwargs:
+                if not is_valid_port(proxy_dict.get(kwargs["proxy_port"])):
+                    logger.error("Invalid proxy port provided.")
+                    raise InvalidPortError("The provided port is not valid.")
+            if "proxy_host" in kwargs:
+                if not is_valid_hostname(proxy_dict.get(kwargs["proxy_host"])):
+                    logger.error("Invalid proxy host provided.")
+                    raise InvalidHostnameError("The provided hostname is not valid.")
```

Comment:
so we validate port and host only when user will provide specific parameters in kwargs? What about the default ones? Maybe we can get rid of kwargs, since they are not used anywhere but here, and add those two as a named arguments, what do you think?